### PR TITLE
Add testing guidelines

### PR DESCRIPTION
**Pull request by german664**

### CODE REVIEW

File: docs/testing/unit-tests/general_testing_guidelines.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+- Autocomplete for mock methods
+- Compile-time type checking
+- Better refactoring support
+
+## Workflow
+
+### When writing tests for hooks
+
+1. Always follow `.agents/testing-hooks-guide.md`
```
Comment (glad-adyen) [inline docs/testing/unit-tests/general_testing_guidelines.md]: Change this testing guide reference to reflect the correct file.

File: docs/testing/unit-tests/general_testing_guidelines.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+[If 100% is NOT achievable]
+Maximum achievable coverage: X%
+Reason 100% is not possible: [explanation]
+Recommended: Accept X% coverage because [reasoning]
+```
+
+## Coverage Requirements
+
+- 85% minimum coverage
```
Comment (glad-adyen) [inline docs/testing/unit-tests/general_testing_guidelines.md]: This minimum coverage requirement could conflict with the achievable coverage expectations set earlier.

File: docs/testing/unit-tests/hooks_testing_guidelines.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+import useHook from './useHook';
+
+describe('useHook', () => {
+    const mockUseCoreContext = vi.mocked(useCoreContext);
+
+    beforeEach(() => {
+        vi.useFakeTimers();
+        vi.setSystemTime(1703520645123);
+        vi.clearAllMocks();
+        mockUseCoreContext.mockReturnValue({ i18n: new Localization().i18n } as any);
```
Comment (glad-adyen) [inline docs/testing/unit-tests/hooks_testing_guidelines.md]: Remove the `as any` assertion here.

File: docs/testing/unit-tests/hooks_testing_guidelines.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    expect(undefinedResult.current.value).toBe(expectedFormat);
+    expect(noParamResult.current.value).toBe(expectedFormat);
+});
+```
+
+### Code Style
+
+- Use consistent formatting with trailing commas
+- Simple arrays stay inline, complex ones get expanded
+- Keep variable names descriptive and purpose-driven
```
Comment (glad-adyen) [inline docs/testing/unit-tests/hooks_testing_guidelines.md]: This section should be move to the general testing guidelines.

File: docs/testing/unit-tests/hooks_testing_guidelines.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+## Quality Checklist
+
+- Analysis phase completed and confirmed
+- All assertions use concrete values (no typeof, toMatch, toContain)
+- Testing behavior, not implementation
+- Common setup in beforeEach, test-specific constants in test functions
+- Edge cases covered with concrete expectations
+- Async operations properly awaited
+- Mocks cleared between tests
+- Coverage meets 85% minimum
```
Comment (ElifDogaliBal) [inline docs/testing/unit-tests/hooks_testing_guidelines.md]: On this checklist I think we can add 2 more points:
 - Follow the best practices for vitest
- Try to cover more with fewer and more precise tests. Avoid having too many tests if it is not increasing the coverage.

[the remark this answers is not in the document]
Comment (german664) [reply docs/testing/unit-tests/hooks_testing_guidelines.md]: ✅

[the remark this answers is not in the document]
Comment (german664) [reply docs/testing/unit-tests/hooks_testing_guidelines.md]: ✅

File: docs/testing/unit-tests/general_testing_guidelines.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+- Focus on behavior coverage, not just line coverage
+- Accept build artifact gaps (HMR, source maps)
+- If coverage is less than 80%, iterate
```
[the remark this answers is not in the document]
Comment (german664) [reply docs/testing/unit-tests/general_testing_guidelines.md]: ✅

File: docs/testing/unit-tests/general_testing_guidelines.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+### When writing tests for hooks
+
+1. Always follow `.agents/testing-hooks-guide.md`
+2. Complete the analysis phase before writing any tests
+3. Wait for user confirmation after analysis
+
+### When writing tests for components
+
+1. Always follow `.agents/testing-components-guide.md`
```
[the remark this answers is not in the document]
Comment (german664) [reply docs/testing/unit-tests/general_testing_guidelines.md]: ✅

File: docs/testing/unit-tests/general_testing_guidelines.md (hunk 1 above)
Comment (german664) [reply docs/testing/unit-tests/general_testing_guidelines.md]: ✅

File: .agents/testing/unit-tests/hooks_testing_guidelines.md [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+## Anti-Patterns
+
+### Avoid Test Repetition
+
+Don't create multiple tests for the same behavior. Consolidate:
+
+```typescript
+// Good: Single test with data-driven approach
+test('should handle nullish timezones gracefully', () => {
+    const { result: nullResult } = renderHook(() => useHook(null as any));
```
Comment (glad-adyen) [inline .agents/testing/unit-tests/hooks_testing_guidelines.md]:
```suggestion
    const { result: nullResult } = renderHook(() => useHook(null));
```
